torch/csrc/PyInterpreter.cpp

```diff
@@ -18,17 +18,24 @@ namespace {
 // because passing in constexpr char* as template argument breaks some
 // versions of MSVC that are being used internally at Meta.
 // MSVC 14.16.27023 (vs2017_15.9)
-#define CONCRETE_TRACE_CUDA(func_name, ...)                           \
-  at::impl::MaybeSetTLSOnEntryGuard guard;                            \
-  if (Py_IsInitialized()) {                                           \
-    pybind11::gil_scoped_acquire gil;                                 \
-    try {                                                             \
-      py::module mod = py::module::import("torch.utils._cuda_trace"); \
-      py::object hook = mod.attr(func_name).attr("fire_callbacks");   \
-      hook(__VA_ARGS__);                                              \
-    } catch (const std::exception& e) {                               \
-      LOG(ERROR) << "CUDA trace hook execution failed: " << e.what(); \
-    }                                                                 \
+#define CONCRETE_GPU_TRACE(device_type, func_name, ...)                       \
+  at::impl::MaybeSetTLSOnEntryGuard guard;                                    \
+  if (Py_IsInitialized()) {                                                   \
+    pybind11::gil_scoped_acquire gil;                                         \
+    try {                                                                     \
+      /* Masquerade hip as cuda because hip uses `torch.cuda` module. */      \
+      if (device_type == at::kHIP) {                                          \
+        device_type = at::kCUDA;                                              \
+      }                                                                       \
```

Comment:
@albanD In my opinion here needs your review again. The code change is that we need to specifically handle `hip` because `hip` uses `torch.cuda` module instead of `torch.hip`.

Reply:
All CI failures seem unrelated.

Reply:
Ho that is pretty sad... Thanks for the update!